src/content/reference/react/cache.md

```diff
@@ -0,0 +1,443 @@
+
+`cache` returns a cached version of `fn` with the same type signature.
+
+When calling `cachedFn` with given arguments, it first checks if a cached result exists in the cache. If a cached result exists, it returns the result. If not, it calls `fn` with the arguments, stores the result in the cache, and returns the result.
+
+<Note>
+
+The optimization of caching return values based on inputs is known as [_memoization_](https://en.wikipedia.org/wiki/Memoization). We refer to `cachedFn` as a memoized function.
+
+</Note>
+
+#### Caveats {/*caveats*/}
+
+[//]: # 'TODO: add links to Server/Client Component reference once https://github.com/reactjs/react.dev/pull/6177 is merged'
+
+- `cache` is recommended for Server Components only. There are plans to introduce `cache` for Client Components, but it is not recommended today.
+- React will invalidate the cache for a memoized function across server requests.
+- Each call to `cache` creates a new function. This means that wrapping `cache` around the same function multiple times will return different memoized functions and they will not share the same cache.
+- The benefit of `cache` is to skip duplicate work by sharing a cache. To promote cache sharing, `cachedFn`, should be defined in a scope that is accessible to multiple components. In most cases, this means calling `cache` and defining the memoized function in the global scope.
+- Cache access only occurs during a component render. This means a call to `cachedFn` outside of a Server Component will not update or read the cache.
+- `cachedFn` will also cache errors. If `fn` throws an error for certain arguments, it will be cached, and re-thrown when `cachedFn` is called with those arguments.
+
+---
+
+## Usage {/*usage*/}
+
+### Share a snapshot of data {/*take-and-share-snapshot-of-data*/}
+
+To share a snapshot of data between components, wrap `cache` around a data-fetching function like `fetch`. When multiple components make the same data fetch, only one request is made and the data returned is cached and shared across components. All components refer to the same snapshot of data across the server render. 
+
+```js [[1, 4, "city"], [1, 5, "fetchTemperature(city)"], [2, 4, "getTemperature"], [2, 9, "getTemperature"], [1, 9, "city"], [2, 14, "getTemperature"], [1, 14, "city"]]
+import {cache} from 'react';
+import {fetchTemperature} from './api.js';
+
+const getTemperature = cache(async (city) => {
+	return await fetchTemperature(city);
+});
+
+async function AnimatedWeatherCard({city}) {
+	const temperature = await getTemperature(city);
+	// ...
+}
+
+async function MinimalWeatherCard({city) {
+	const temperature = await getTemperature(city);
+	// ...
+}
+```
+
+If `AnimatedWeatherCard` and `MinimalWeatherCard` both render for the same <CodeStep step={1}>city</CodeStep>, they will receive the same snapshot of data from the <CodeStep step={2}>memoized function</CodeStep>. The <CodeStep step={1}>city</CodeStep> acts as a cache key.
```

Comment:
It might be helpful to call out that `fetchTemperature` is not called until the first time that `getTemperature` is called.

---

> The `city` acts as a cache key.

Maybe before this sentence, explicitly mention that…

> If `AnimatedWeatherCard` and `MinimalWeatherCard` supply different `city` arguments to `getTemperature`, then `fetchTemperature` will be called twice and each call site will receive different data.

Reply:
Re: second point, added!